fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/OCFLPersistentStorageUtils.java

```diff
@@ -74,26 +79,52 @@ private OCFLPersistentStorageUtils() {
     private static final String FEDORA_METADATA_SUFFIX = "/" + FedoraTypes.FCR_METADATA;
 
     /**
-     * Returns the relative subpath of the resourceId based on the ancestor's resource id.
+     * Returns the relative subpath of the resourceId based on the root object's resource id.
      *
-     * @param ancestorResourceId The ancestor resource
-     * @param resourceId         The identifier of the resource whose subpath you wish to resolve.
+     * @param rootObjectId The fedora root object identifier
+     * @param resourceId   The identifier of the resource whose subpath you wish to resolve.
      * @return The resolved subpath
      */
-    public static String relativizeSubpath(final String ancestorResourceId, final String resourceId) {
-        if (resourceId.equals(ancestorResourceId)) {
-            return resourceId;
-        } else if (resourceId.startsWith(ancestorResourceId)) {
-            return resourceId.substring(ancestorResourceId.length() + 1);
+    public static String relativizeSubpath(final String rootObjectId, final String resourceId) {
+        if (resourceId.equals(rootObjectId)) {
+            return resourceId.substring(FEDORA_ID_PREFIX.length());
+        } else if (resourceId.startsWith(rootObjectId)) {
+
+            final var rawSubpath = resourceId.substring(rootObjectId.length() + 1);
+            //grab the last path segment of the rootObjectId
+            final var lastPathSegment = rootObjectId.substring(rootObjectId.lastIndexOf("/") + 1);
+
+            if (rawSubpath.endsWith(FCR_ACL) || rawSubpath.endsWith(FCR_METADATA)) {
```

Comment:
Kind of wondering, could this logic be accomplished with a string replace, replacing `FCR_ACL` => `-description`, etc?

Reply:
I'm wondering how this works with things like
```
info:fedora/some/binary/fcr:metadata/fcr:versions
```
or are we stripping off the versions endpoint before here?

But yes I think something like this could work (if I understand the purpose)
```
if (rawSubpath.endsWith(FCR_ACL)) {
  return lastPathSegment + rawSubpath.replace("/" + FCR_ACL, "-acl");
} else if (rawSubpath.endsWith(FCR_METADATA)) {
  return lastPathSegment + rawSubpath.replace("/" + FCR_METADATA, "-description");
} else {
  return lastPathSegment + "/" +  rawSubpath;
}
```

Reply:
Good question about `fcr:versions`. It probably gets stripped off before this point since it goes through a different controller and has to be resolved into requesting a timemap rather than a file on disk.

Reply:
Yes - that's my understanding too.